site/source/tutorials/reactive_programming.md

```diff
@@ -0,0 +1,58 @@
+
+Consider a banking application where the user's account balance might show up in several different places on the screen. After the user makes a withdrawl, we want the account balance to update.
+
+## Imperative Approach
+
+Traditionally, we might use an event emitter. Every time balance is updated we emit an event, and any interested parties can listen in.
+
+![Banking](resources/imperative.png)
+
+```javascript
+// assume some sort of application wide MessageBus
+
+function makeWithdrawl() {
+    // make the withdrawl
+    messageBus.emit('account-balance-updated', newBalance);
+}
+
+// at different points in the UI
+messageBus.on('account-balance-updated', (newBalance) => {
+    accountDomElement.innerText = newBalance;
+});
+```
+
+With an event emitter, any interested widget can listen for updates to the account balance.  This pattern has served well in the past, but there are a few problems with this approach:
+
+*   We need to remember to emit `account-balance-updated` every time the account balance is changed.
+*   We need to rememer to listen for balance updates.
+*   We need to have a facitlity to **remove** the event listeners when they are no longer needed.
+
+## Reactive Approach
+
+Dojo2 is built around unidirectional, top-down ,property propagation It is the parent widget's job to pass properties down to it's children. In fact, a child widget has no way to reference a parent widget! When a property changes, widgets are re-rendered (using an efficient virtual DOM) to reflect the updated state. 
+
+In our banking example, we simply provide the account balance as a property to the widget, and when the property changes, so does our UI. The account balance is propagated down to the widgets that need it.
+
+![Account Balance](resources/dojo2-reactive.png)
+
+```typescript
+export interface AccountBalanceProperties extends WidgetProperties {
+    balance: number;
+}
+
+class AccountBalance extends WidgetBase<AccountBalanceProperties> {
+    render() {
+      	const { balance } = this.properties;
+        return v('span', {}, [ balance ]);
+    }
+}
+```
+
+Our `AccountBalance` widget is only concerned with it's `balance` property, and that property is provided by the parent widget.  Setting the balance property at the application level will cause the balance to be updated across the entire application.
+
+## What's Next?
+
+Reactive programming patterns are being used in Dojo2 to make widgets consistent and easy to author.  Want to learn more about reactive programming? There are a number of additional reactive patterns that can be used to manage your application state (like [Redux](http://redux.js.org/), or [Observables](http://reactivex.io/documentation/observable.html)).
```

Comment:
May want to add a sentence that says that Dojo 2 will also soon have a reactive application store that is expected to arrive for beta 3?